velox/functions/prestosql/DateTimeFunctions.h

```diff
@@ -118,23 +118,25 @@ FOLLY_ALWAYS_INLINE bool call(
 }
 VELOX_UDF_END();
 
-VELOX_UDF_BEGIN(hour)
-const date::time_zone* timeZone_ = nullptr;
+struct udf_hour {
+  VELOX_UDF_BEGIN2(hour)
```

Comment:
it's a shame you have to repeat the "hour" here.  I imagine confusing things happen if you get this out of sync with the name above?  Is there some way of removing that.  Looking at VELOX_UDF_BEGIN2 it only seems to use it in one place to define a string for `static constexpr auto name`.  

Another variant that might spark an even better idea, you could get make the struct visible and not repeat the name using something like:

struct VELOX_UDF("hour") {
   ...
}

you don't even need the VELOX_UDF_END as it's just a } and it's obvious you need it and what scope it encloses.

Reply:
actually ignore that, I just realized that won't work, the user supplied code will be in an extra level of nesting compared to what VELOX_UDF expands to. :(